homeassistant/components/smartthings/__init__.py

```diff
@@ -35,6 +39,33 @@
     return True
 
 
+async def async_migrate_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Handle migration of a previous version config entry.
+
+    A config entry created under a previous version must go through the
+    integration setup again so we can properly retrieve the needed data
+    elements. Force this by removing the entry and triggering a new flow.
+    """
+    from pysmartthings import SmartThings
+
+    # Delete the installed app
+    api = SmartThings(async_get_clientsession(hass),
+                      entry.data[CONF_ACCESS_TOKEN])
+    await api.delete_installed_app(entry.data[CONF_INSTALLED_APP_ID])
+    # Delete the entry
+    hass.async_create_task(
+        hass.config_entries.async_remove(entry.entry_id))
+    # only create new flow if there isn't a pending one for SmartThings.
+    flows = hass.config_entries.flow.async_progress()
+    if not [flow for flow in flows if flow['handler'] == DOMAIN]:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={'source': 'import'}))
+
+    # Return False because it could not be migrated.
+    return False
```

Comment:
Won't this log an error? It's weird if we error on expected behavior.

Reply:
No -- it will only [log an error if an exception is raised or bool wasn't returned](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/config_entries.py#L382).  Returning `False` stops the setup operation so it doesn't go on to try and call `async_setup_entry`.